lib/mongoid_rails_migrations/mongoid_ext/railties/database.rake

```diff
@@ -1,71 +1,124 @@
+      task :up => :environment do
+        version = ENV["VERSION"] ? ENV["VERSION"].to_i : nil
+        raise "VERSION is required" unless version
+        Mongoid::Migrator.run(:up, Mongoid::Migrator.migrations_path, version)
+      end
+
+      desc 'Runs the "down" for a given migration VERSION.'
+      task :down => :environment do
+        version = ENV["VERSION"] ? ENV["VERSION"].to_i : nil
+        raise "VERSION is required" unless version
+        Mongoid::Migrator.run(:down, Mongoid::Migrator.migrations_path, version)
+      end
+
+      desc 'Display status of migrations'
+      task :status => :environment do
+        Mongoid::Migrator.status(Mongoid::Migrator.migrations_path)
+      end
+    end
   end
 
-  desc 'Drops the database for the current Mongoid client'
-  task :drop => :environment do
-    # Unlike Mongoid's default, this implementation supports the MONGOID_CLIENT_NAME override
-    Mongoid::Migration.connection.database.drop
+  unless Rake::Task.task_defined?("db:drop")
+    desc '[DEPRECATED] Use "db:mongoid:drop" instead.'
+    task :drop => "db:mongoid:drop"
   end
 
-  desc 'Current database version'
-  task :version => :environment do
-    puts Mongoid::Migrator.current_version.to_s
+  unless Rake::Task.task_defined?("db:version")
+    desc '[DEPRECATED] Use "db:mongoid:version" instead.'
+    task :version => "db:mongoid:version"
   end
 
-  desc "Migrate the database through scripts in db/migrate. Target specific version with VERSION=x. Turn off output with VERBOSE=false."
-  task :migrate => :environment do
-    Mongoid::Migration.verbose = ENV["VERBOSE"] ? ENV["VERBOSE"] == "true" : true
-    Mongoid::Migrator.migrate(Mongoid::Migrator.migrations_path, ENV["VERSION"] ? ENV["VERSION"].to_i : nil)
+  unless Rake::Task.task_defined?("db:rollback")
+    desc '[DEPRECATED] Use "db:mongoid:rollback" instead.'
+    task :rollback => "db:mongoid:rollback"
   end
 
-  namespace :migrate do
-    desc 'Rollback the database one migration and re migrate up. If you want to rollback more than one step, define STEP=x. Target specific version with VERSION=x.'
-    task :redo => :environment do
-      if ENV["VERSION"]
-        Rake::Task["db:migrate:down"].invoke
-        Rake::Task["db:migrate:up"].invoke
-      else
-        Rake::Task["db:rollback"].invoke
-        Rake::Task["db:migrate"].invoke
-      end
-    end
+  unless Rake::Task.task_defined?("db:rollback_to")
+    desc '[DEPRECATED] Use "db:mongoid:rollback_to" instead.'
+    task :rollback_to => "db:mongoid:rollback_to"
+  end
 
-    desc 'Resets your database using your migrations for the current environment'
-    # should db:create be changed to db:setup? It makes more sense wanting to seed
-    task :reset => ["db:drop", "db:create", "db:migrate"]
+  unless Rake::Task.task_defined?("db:migrate")
+    desc '[DEPRECATED] Use "db:mongoid:migrate" instead.'
+    task :migrate => "db:mongoid:rollback_to"
+  end
```

Comment:
Wrong alias here I guess ? Why is this pointing to `rollback_to` instead of `migrate` ?
About the issue you raised with `db:migrate` being already defined by Mongoid as a noop (so we can't use it to detect the presence of AR), I would suggest another simpler option then: You can use the `Rake::Task.task_defined?("db:rollback")` condition maybe (this one is not defined by mongoid) and include the definition of other tasks like `migrate` inside the same condition ?